django/http/multipartparser.py

```diff
@@ -682,21 +683,36 @@ def parse_boundary_stream(stream, max_header_size):
     """
     Parse one and exactly one stream that encapsulates a boundary.
     """
-    # Stream at beginning of header, look for end of header
-    # and parse it if found. The header must fit within one
-    # chunk.
-    chunk = stream.read(max_header_size)
 
-    # 'find' returns the top of these four bytes, so we'll
-    # need to munch them later to prevent them from polluting
-    # the payload.
-    header_end = chunk.find(b"\r\n\r\n")
+    # Start with checking for the end of headers in the first kilobyte and if not found
+    # extend the search to double the size (max 8192 bytes)
+    headers_chunk_size = 1024
+    while True:
+        if headers_chunk_size > max_header_size:
+            raise MultiPartParserError("Request max total header size exceeded.")
```

Comment:
This is untested.

Reply:
Yeah, only the effects of it in an 'e2e' scenario in `test_upload_header_fields_too_large`.

Going to add a test for the exception explicitly.

Reply:
Added